frontend/src/component/admin/groups/GroupForm/GroupForm.tsx

```diff
@@ -83,23 +105,47 @@ export const GroupForm: FC<IGroupForm> = ({
     description,
     mappingsSSO,
     users,
+    rootRole,
     setName,
     setDescription,
     setMappingsSSO,
     setUsers,
     handleSubmit,
     handleCancel,
+    setRootRole,
     errors,
     mode,
     children,
 }) => {
     const { config: oidcSettings } = useAuthSettings('oidc');
     const { config: samlSettings } = useAuthSettings('saml');
+    const { uiConfig } = useUiConfig();
+    const { roles } = useUsers();
 
     const isGroupSyncingEnabled =
         (oidcSettings?.enabled && oidcSettings.enableGroupSyncing) ||
         (samlSettings?.enabled && samlSettings.enableGroupSyncing);
 
+    const groupRootRolesEnabled = Boolean(uiConfig.flags.groupRootRoles);
+
+    const roleIdToRole = (rootRoleId: number | null): IProjectRole | null => {
+        return (
```

Comment:
This little dance is because the `StyledAutocompleteWrapper` does not tolerate `undefined` correctly. It very much wants a `null`

Reply:
Yep, but like I mentioned in https://github.com/Unleash/unleash/pull/3559/files#r1171626519 I think if it's `undefined` you're already covering your bases, since it's going to be `null` here. Up to you if you'd like to try it out in case you feel like it's more straightforward.

Reply:
Yeah *sigh* I did try. It doesn't work as well as I'd like. The find returns `undefined` if it uh... doesn't find anything. It does work but it splashes a huge warning to the console. Not a fan of warnings there since they tend to hide things we really should care about